secrethub/resource_access_rule.go

```diff
@@ -33,10 +33,35 @@ func resourceAccessRule() *schema.Resource {
 				Description: "The permission that the account has on the given directory: read, write or admin",
 			},
 		},
+		Importer: &schema.ResourceImporter{
+			State: resourceAccessRuleImport,
+		},
+	}
+}
+
+func resourceAccessRuleCreate(d *schema.ResourceData, m interface{}) error {
+	provider := m.(providerMeta)
+	client := *provider.client
+
+	path := d.Get("dir_path").(string)
+	permission := d.Get("permission").(string)
+	account := d.Get("account_name").(string)
+
+	_, err := client.AccessRules().Get(path, account)
+	if err == nil {
+		return fmt.Errorf("access rule already exists: %s:%s", path, account)
+	} else if err != api.ErrAccessRuleNotFound {
+		return err
 	}
+
+	_, err = client.AccessRules().Set(path, permission, account)
+
+	d.SetId(path + ":" + account)
```

Comment:
If `Set` failed, we would want to assign it an id.

Reply:
We do

Reply:
The `err` return is below this `SetId`

Reply:
we wouldn't*

Reply:
Yes, you are right

https://www.terraform.io/docs/extend/writing-custom-providers.html#error-handling-amp-partial-state

>If the Create callback returns with or without an error and an ID has been set, the resource is assumed created and all state is saved with it. Repeating because it is important: if there is an error, but the ID is set, the state is fully saved.

Let's pick this up in a new PR, as this problem isn't introduced in this PR and is broader than what is touched in this PR (it's also in the update func)